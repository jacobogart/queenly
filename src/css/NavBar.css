.NavBar {
  align-items: center;
  display: flex;
  height: 50px;
  justify-content: space-between;
  margin: 0 auto;
  padding: 0.5rem;
  width: 95%;
  overflow: hidden;
}

.navLogo {
  font-family: "Lobster";
  color: lightgray;
  font-size: 2.2em;
  padding: 5px 0 0 20px;
}

.NavBar .searchBarContainer {
  display: flex;
  z-index: 9;
  height: 500%;
  margin-top: 210px;
}

.NavBar .searchHolder {
  position: absolute;
  z-index: 10;
  width: 533px;
  margin-top: 40px;
  overflow: auto;
}

.searchHolder {
  width: 500px;
}

i {
  color: lightgrey;
}

i:hover {
  color: white;
}
