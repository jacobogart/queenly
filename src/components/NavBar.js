import React from "react";
import Hamburger from "./Hamburger";
import Favorites from "./Favorites";
import "../css/NavBar.css";
import SearchBar from "./SearchBar";

class NavBar extends React.Component {
  constructor(props) {
    super(props);
  }

  displaySplashPage = () => {
    this.props.displaySplashPage();
  };

  render() {
    if (!this.props.showSplashPage) {
      var navLogo = (
        <h2 className="navLogo" onClick={this.displaySplashPage}>
          Queenly
        </h2>
      );
      var searchBar = <SearchBar {...this.props} />;
    }

    return (
      <nav className="NavBar">
        <Hamburger {...this.props} />
        {navLogo}
        {searchBar}
        <Favorites {...this.props} />
      </nav>
    );
  }
}

export default NavBar;
