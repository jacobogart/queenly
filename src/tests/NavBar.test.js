import React from "react";
import NavBar from "../components/NavBar";
import { shallow } from "enzyme";

//navbar should exist
describe("<NavBar />", () => {
  it("should exist", () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper.exists()).toBe(true);
  });
  it("should match the snapshot", () => {
    const tree = shallow(<NavBar />);
    expect(tree).toMatchSnapshot();
  });
  it("should render two elements on splash page", () => {
    const wrapper = shallow(<NavBar showSplashPage={true} />);
    expect(wrapper.find("nav").children().length).toBe(2);
  });
  it("should only have hamburger and favorites element while on splash page", () => {
    const wrapper = shallow(<NavBar showSplashPage={true} />);
    //may need to change these selectors
    expect(wrapper.find("Hamburger").exists()).toBe(true);
    expect(wrapper.find("Favorites").exists()).toBe(true);
    expect(wrapper.find("SearchBar").exists()).toBe(false);
  });
  it("should render three elements while not on splash page", () => {
    const wrapper = shallow(<NavBar showSplashPage={false} />);
    expect(wrapper.find("nav").children().length).toBe(3);
  });
  it("should render three elements: [hamburger, searchbar, favorites] while not on splash page", () => {
    const wrapper = shallow(<NavBar showSplashPage={false} />);
    expect(wrapper.find("Hamburger").exists()).toBe(true);
    expect(wrapper.find("Favorites").exists()).toBe(true);
    expect(wrapper.find("SearchBar").exists()).toBe(true);
  });
});
