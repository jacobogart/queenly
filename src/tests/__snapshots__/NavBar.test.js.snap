// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match the snapshot 1`] = `
<nav
  className="NavBar"
>
  <Hamburger />
  <SearchBar />
  <Favorites />
</nav>
`;
